src/web/client/common/remoteServiceProvider.ts

```diff
@@ -0,0 +1,147 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+'use strict';
+import * as vscode from 'vscode';
+import { getHeader } from './authenticationProvider';
+import { DEFAULT_LANGUAGE_CODE, pathparam_schemaMap, SINGLE_ENTITY_SAVE_URL_KEY, WEBPAGES_FILENAME } from './constants';
+import { CONTENTPAGES, PORTALSFOLDERNAME, PORTALSURISCHEME, SINGLE_ENTITY_FETCH_URL_KEY, WEBPAGES, WEBTEMPLATES } from './constants';
+import { PortalsFS } from './fileSystemProvider';
+import { entitiesSchemaMap, dataSourcePropertiesMap } from './localStore';
+
+export async function fetchData(accessToken: string, entity: string, entityId: string, queryParamsMap: any, entitiesSchemaMap: any, languageIdCodeMap: any, portalFs: PortalsFS) {
+    try {
+        const dataverseOrg = queryParamsMap.get('orgUrl');
+        const requestUrl = getRequestURLSingleEntity(dataverseOrg, entity, entityId, SINGLE_ENTITY_FETCH_URL_KEY, entitiesSchemaMap);
+        vscode.window.showInformationMessage(requestUrl);
+        const response = await fetch(requestUrl, {
+            headers: getHeader(accessToken),
+        });
+        if (!response.ok) {
+            vscode.window.showInformationMessage("auth failed in fetch data");
+            throw new Error(response.statusText);
+        }
+        const data = await response.json();
+        switch (entity) {
```

Comment:
we should not have any of these. Only for exception cases we should handle differently.